core/dbt/include/global_project/macros/materializations/helpers.sql

```diff
@@ -72,3 +72,96 @@
+     {{ return(on_schema_change) }}
+   
+   {% endif %}
+
+{% endmacro %}
+
+{% macro get_column_names(columns) %}
+  {% set result = [] %}
+  
+  {% for col in columns %}
+    {{ result.append(col.column) }}
+  {% endfor %}
+  
+  {{ return(result) }}
+{% endmacro %}
+
+{% macro diff_columns(source_columns, target_columns) %}
+  {% set result = [] %}
+  {% set source_names = get_column_names(source_columns) %}
+  {% set target_names = get_column_names(target_columns) %}
+   
+   {# check whether the name attribute exists in the target, but dont worry about data type differences #}
+   {%- for col in source_columns -%} 
+     {%- if col.name not in target_names -%}
+      {{ result.append(col) }}
+      {%- endif -%}
+   {%- endfor -%}
+  
+  {{ return(result) }}
+{% endmacro %}
+
+{% macro check_for_schema_changes(source_relation, target_relation) %}
+  
+  {% set schema_changed = False %}
+  {%- set source_columns = adapter.get_columns_in_relation(source_relation) -%}
+  {%- set target_columns = adapter.get_columns_in_relation(target_relation) -%}
+  {%- set source_not_in_target = diff_columns(source_columns, target_columns) -%}
+  {%- set target_not_in_source = diff_columns(target_columns, source_columns) -%}
+
+  {% if source_not_in_target != [] %}
+    {% set schema_changed = True %}
+  {% elif target_not_in_source != [] %}
+    {% set schema_changed = True %}
+  {% endif %}
+
+  {{ return(schema_changed) }}
+
+{% endmacro %}
+
+{% macro sync_columns(source_relation, target_relation, on_schema_change='append') %}
+  
+  {%- set source_columns = adapter.get_columns_in_relation(source_relation) -%}
+  {%- set target_columns = adapter.get_columns_in_relation(target_relation) -%}
+  {%- set add_to_target_arr = diff_columns(source_columns, target_columns) -%}
+  {%- set remove_from_target_arr = diff_columns(target_columns, source_columns) -%}
+
+  {%- if on_schema_change == 'append' -%}
+    {%- for col in add_to_target_arr -%}
+       {%- set build_sql = 'ALTER TABLE ' + target_relation.database + '.' + target_relation.schema+'.'+target_relation.name + ' ADD COLUMN ' + col.name + ' ' + col.dtype -%}
+       {%- do run_query(build_sql) -%}
+    {%- endfor -%}
+    
+  {% elif on_schema_change == 'sync' %}
+    {%- for col in add_to_target_arr -%}
+       {%- set build_sql = 'ALTER TABLE ' + target_relation.database + '.' + target_relation.schema+'.'+target_relation.name + ' ADD COLUMN ' + col.name + ' ' + col.dtype -%}
+       {%- do run_query(build_sql) -%}
+    {%- endfor -%}
+
+    {%- for col in remove_from_target_arr -%}
+      {%- set build_sql = 'ALTER TABLE ' + target_relation.database + '.' + target_relation.schema+'.'+target_relation.name + ' DROP COLUMN ' + col.name -%}
+      {%- do run_query(build_sql) -%}
+    {%- endfor -%}
+  
+  {% endif %}
+
+  {{ 
+      return(
+             {
+              'columns_added': add_to_target_arr,
+              'columns_removed': remove_from_target_arr
+             }
+          )
+  }}
```

Comment:
I like the idea of logging some of this information (to debug, not to info/stdout). Something roughly like:
```
Schema change detected. dbt will {{ on_schema_change }} by adding {{ columns_added }} and removing {{ columns_removed }}.
```

Reply:
Good call.